src/test/java/hudson/plugins/claim/UserClaimsReportTest.java

```diff
@@ -73,7 +74,26 @@ public void userAssignedFailedJobsShouldBeVisibleInUserClaimReport() throws Exce
         verifyUserClaims(userA, 1, 0);
         verifyUserClaims(userB, 0, 0);
         verifyUserClaims(userC, 2, 0);
-}
+    }
+
+    @Test
+    public void userAssignedFailedJobsUnderADifferentIdForSameUserShouldBeVisibleInUserClaimReport() throws Exception {
+        long timestamp = System.currentTimeMillis();
+
+        FreeStyleProject job1 = createFailingJobWithName("job1-" + timestamp);
+
+        User userA = User.get("userA-" + timestamp, true, Collections.EMPTY_MAP);
+        assert !userA.getId().equals(userA.getId().toLowerCase()); // ensure id of user is not equal to its lowercase version
```

Comment:
`assert` is only enabled when the -ea flag is activated, furthermore it doesn't give any helpful message in case of failure. Using some `assertEquals` method is always better for unit tests.

Reply:
The goal was to separate the actual test assertion from test setup requirement.
You are however right, would be better en ensure the issue is reported in all cases. 
Will change this.